MyCryptoHQ/MyCrypto
# Allow 0 ETH Transactions

Closes #1275

### Description

Allows for zero ETH transactions on the send tab. Does not allow for zero token transactions, or zero value transaction requests.

### Changes

* Adjusted `validNumber` validator to allow for `>= 0`
* Added `validPositiveNumber` for cases that should be `> 0`
* Updated saga logic to selectively use those two based on whether or not it's an ETH transaction
* Added tests to validate zero eth transaction is valid, zero token transactions are not

### Steps to Test

1. Run app with the `allow-zero` branch
2. Unlock a wallet with tokens on `/send`
3. Enter `0` as the value, confirm form allows for this (generate the tx)
4. Change unit to a token, confirm field becomes invalid
5. Change value to a positive number, confirm field becomes valid again
